src/lib/form-field/form-field.scss

```diff
@@ -145,6 +145,8 @@ $mat-form-field-underline-height: 1px !default;
   position: absolute;
   height: $mat-form-field-underline-height;
   width: 100%;
+  // Prevents underline from disappearing at lower zoom levels.
+  transform: perspective(1px);
```

Comment:
I'm not sure whether this fixes the issue. The only way I managed to break it on my machine was to zoom out to 33% and these changes don't seem to help in that case.

Reply:
Its a little weird, they don't all seem to break, I think it has to do with their vertical position or something. I saw some of them broken at 90% on the input demo page, this fixes it for >= 50% which I think is reasonable

Reply:
I think that it all depends on your OS/Screen size/Browser combo and whether it hits some weird value that doesn't round off correctly. We could get this in and see if people still see the issue.

Reply:
Yeah, I suspect this will fix it, it does for me